File: backend/routers/signals.py
from fastapi import APIRouter, Query

from backend.services.signals import get_config, get_signals_data

router = APIRouter(tags=["signals"])


@router.get("/config")
def signals_config():
    return get_config()


@router.get("/data")
def signals_data(
    project: str = Query(...),
    date: str = Query(...),
    signals: str = Query(...),
):
    return get_signals_data(project, date, signals.split(","))

File: frontend/src/api/signals.ts
import { apiClient } from './client'

export async function fetchSignalsConfig() {
  const { data } = await apiClient.get('/api/signals/config')
  return data
}

export async function fetchSignalsData(project: string, date: string, signals: string[]) {
  const { data } = await apiClient.get('/api/signals/data', {
    params: { project, date, signals: signals.join(',') },
  })
  return data
}
